processor/otel/metrics.go

```diff
@@ -89,6 +90,168 @@ func (c *Consumer) convertResourceMetrics(resourceMetrics pdata.ResourceMetrics,
+							b.nonIdleCPUUtilizationSum.value += sample.Value
+							b.nonIdleCPUUtilizationSum.timestamp = dp.Timestamp().AsTime()
+						}
+					}
+					if cpuIdStr, exists := dp.Attributes().Get("cpu"); exists {
+						cpuId, _ := strconv.Atoi(cpuIdStr.StringVal())
+						if cpuId+1 > b.cpuCount {
+							b.cpuCount = cpuId + 1
+						}
+					}
+				case "runtime.jvm.memory.area":
+					var key jvmMemoryKey
+					dp.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+						switch k {
+						case "area":
+							key.area = v.AsString()
+						case "type":
+							key.type_ = v.AsString()
+						}
+						return true
+					})
+					if key.area != "" && key.type_ != "" {
+						b.jvmMemory[key] = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+					}
+				}
+			}
+		}
+	}
+}
+
+// emit upserts Elastic APM metrics into ms from information accumulated in b.
+func (b *apmMetricsBuilder) emit(ms metricsets) {
+	// system.memory.actual.free
+	// Direct translation of system.memory.usage (state = free)
+	if b.freeMemoryBytes.value > 0 {
+		ms.upsertOne(
+			b.freeMemoryBytes.timestamp, "system.memory.actual.free", pdata.NewAttributeMap(),
+			model.MetricsetSample{Value: b.freeMemoryBytes.value},
+		)
+	}
+	// system.memory.total
+	// system.memory.usage (state = free) + system.memory.usage (state = used)
+	totalMemoryBytes := b.freeMemoryBytes.value + b.usedMemoryBytes.value
+	if totalMemoryBytes > 0 {
+		ms.upsertOne(
+			b.freeMemoryBytes.timestamp, "system.memory.total", pdata.NewAttributeMap(),
+			model.MetricsetSample{Value: totalMemoryBytes},
+		)
+	}
+	// system.cpu.total.norm.pct
+	// Averaging of non-idle CPU utilization over all CPU cores
+	if b.nonIdleCPUUtilizationSum.value > 0 && b.cpuCount > 0 {
+		ms.upsertOne(
+			b.nonIdleCPUUtilizationSum.timestamp, "system.cpu.total.norm.pct", pdata.NewAttributeMap(),
+			model.MetricsetSample{Value: b.nonIdleCPUUtilizationSum.value / float64(b.cpuCount)},
+		)
+	}
+	// jvm.gc.time
+	// Direct translation of runtime.jvm.gc.time or runtime.jvm.gc.collection
+	for k, v := range b.jvmGCTime {
+		elasticapmAttributes := pdata.NewAttributeMap()
+		elasticapmAttributes.Insert("name", pdata.NewAttributeValueString(k))
+		ms.upsertOne(
+			v.timestamp, "jvm.gc.time", elasticapmAttributes,
+			model.MetricsetSample{Value: v.value},
+		)
+	}
+	// jvm.gc.count
+	// Direct translation of runtime.jvm.gc.count
+	for k, v := range b.jvmGCCount {
+		elasticapmAttributes := pdata.NewAttributeMap()
+		elasticapmAttributes.Insert("name", pdata.NewAttributeValueString(k))
+		ms.upsertOne(
+			v.timestamp, "jvm.gc.count", elasticapmAttributes,
+			model.MetricsetSample{Value: v.value},
+		)
+	}
+	// jvm.memory.<area>.<type>
+	// Direct translation of runtime.jvm.memory.area (area = xxx, type = xxx)
+	for k, v := range b.jvmMemory {
+		ms.upsertOne(
+			v.timestamp, fmt.Sprintf("jvm.memory.%s.%s", k.area, k.type_), pdata.NewAttributeMap(),
+			model.MetricsetSample{Value: v.value},
+		)
```

Comment:
I see you've resolved this, but I can't see where we're creating `java.memory.<area>.pool.<type>` metrics. If you'd like, please leave a TODO and we can address it in a followup.

Otherwise, what we need to do is:
 - in `accumulate`, set the `jvmMemoryKey.pool` attribute
 - here in `emit`, emit either `jvm.memory.<area>.<type>` or `jvm.memory.<area>.pool.<type>`, depending on whether  `jvmMemoryKey.pool` is non-empty

Reply:
Thanks for the heads-up ; I think I addressed the pool issue in [c8129eb](https://github.com/elastic/apm-server/pull/7090/commits/c8129eb277d24fd340a8653acb0c1b5b53c306b7), in an identical fashion to what you describe, although I might have missed the point.

`accumulate`: 
https://github.com/jlvoiseux/apm-server/blob/a47157b0238ed8e887755bea59947753421ecc1f/processor/otel/metrics.go#L178-L189

`emit`: 
https://github.com/jlvoiseux/apm-server/blob/a47157b0238ed8e887755bea59947753421ecc1f/processor/otel/metrics.go#L249-L262

Reply:
Sorry, I must have been looking at a stale commit. Looks perfect!